xonsh/tools.py

```diff
@@ -1358,6 +1359,11 @@ def to_itself(x):
     return x
 
 
+def to_int_or_none(x) -> tp.Optional[int]:
+    """Convert the given value to integer if possible. Otherwise return None"""
+    return int(x) if str(x).isdigit() else None
```

Comment:
Maybe this should be `None if x is None else int(x)`

Reply:
is it possible that the value to be "None" as it is retrieved as str (As I understand)?

Reply:
Oh yeah, it is.... In that case you should check if `x.lower() == "none"`.  Other string values should still result in an error if they are not ints

Reply:
Ok